fem/fe/fe_base.hpp

```diff
@@ -577,6 +592,13 @@ class FiniteElement
    virtual const DofToQuad &GetDofToQuad(const IntegrationRule &ir,
                                          DofToQuad::Mode mode) const;
 
+   /** @brief Return a DofToQuad structure corresponding to the given
+       IntegrationRule using the given DofToQuad::Mode and ElementDofOrdering. */
+   /** See the documentation for DofToQuad and ElementDofOrdering for more details.
+   TODO - make this a reference again.*/
+   virtual const DofToQuad &GetDofToQuad(const IntegrationRule &ir,
+                                         DofToQuad::Mode mode,
+                                         const ElementDofOrdering ordering) const;
```

Comment:
Should `ordering` just be added as an optional argument to `GetDofToQuad` (i.e. with a default value) to avoid having the interface become more complicated?

Reply:
I am not sure it is that easy to do.

Reply:
I think @v-dobrev would prefer that solution, if you have time to look into it, it would be great, but I fear that it might not be that easy to do.

Reply:
I added `LEXICOGRAPHIC_FULL` and removed the overloads for `GetDofToQuad`. I also changed the logical conditions in `MFEM_VERIFY`s that appeared in `GetDofToQuad` to take the new enum value into account.